src/main/scala/org/scanet/math/MathBaseOp.scala

```diff
@@ -131,23 +131,127 @@ import scala.Ordering.Implicits._
 
+  /** Computes the sum of elements across dimensions of a tensor.
+   *
+   * Reduces `out` along the dimensions given in `axises`.
+   * The rank of the tensor is reduced by 1 for each entry in `axises`.
+   *
+   * {{{Tensor.matrix(Array(1, 2, 3), Array(4, 5, 6)).const.sum(Seq(0)).eval should be(Tensor.vector(5, 7, 9))}}}
+   *
+   * @param out tensor
+   * @param axises to sum
+   * @return tensor with summed values
+   */
   def sum[A: TensorType: Numeric](out: F[A], axises: Seq[Int]): F[A]
 
+  /** Computes the sum of elements across all dimensions of a tensor.
+   *
+   * {{{Tensor.matrix(Array(1, 2, 3), Array(4, 5, 6)).const.sum.eval should be(Tensor.scalar(21))}}}
+   *
+   * @param out tensor
+   * @return tensor with summed values
+   */
   def sum[A: TensorType: Numeric](out: F[A]): F[A]
 
+  /** Shuffle dimensions of `out` according to a permutation.
+   *
+   * {{{
+   * val before = Tensor(range(1, 13), Shape(2, 2, 3))
+   * val after = Tensor(Array(1, 4, 2, 5, 3, 6, 7, 10, 8, 11, 9, 12), Shape(2, 3, 2))
+   * before.const.transpose(Seq(0, 2, 1)).eval should be(after)
+   * }}}
+   *
+   * @param out tensor to transpose
+   * @param perm dimensions permutations
+   * @return transposed tensor
+   */
   def transpose[A: TensorType: Numeric](out: F[A], perm: Seq[Int]): F[A]
 
+  /** Transpose given tensor - flip dimensions over its diagonal.
+   *
+   * {{{
+   * val matrix = Tensor.matrix(Array(1, 2), Array(3, 4)).const
+   * matrix.transpose.eval should be(Tensor.matrix(Array(1, 3), Array(2, 4)))
+   * }}}
+   *
+   * @param out tensor to transpose
+   * @return transposed tensor
+   */
   def transpose[A: TensorType: Numeric](out: F[A]): F[A]
 
+  /** Compute decaying average of given value by putting more weight into latest (`next`) value.
+   * It is equivalent to `(prev * decay) + ((1 - decay) * next)`.
+   *
+   * {{{10.0.const.decayingAvg(5.0.const, 0.9.const).eval should be(Tensor.scalar(9.5))}}}
+   *
+   * @param avg previous value
+   * @param next current value
+   * @param decay rate of decay
+   * @return decaying average
+   */
   def decayingAvg[A: TensorType: Numeric](avg: F[A], next: F[A], decay: F[A]): F[A]
 
+  /** Compute root mean squared of given value.
+   * It is equivalent to `(out + epsilon) ^ 0.5`.
+   *
+   * {{{8f.const.rms(1f.const).eval should be(Tensor.scalar(3f))}}}
+   *
+   * @return root mean squared
+   */
   def rms[A: TensorType: Numeric](out: F[A], epsilon: F[A]): F[A]
+
+  /** Remove bias to a component from the given (computed) value
+   * It is equivalent to `out / (1 - bias ^ iter)`
+   *
+   * {{{
+   * val x = 1.5f.const
+   * val bias = 0.5f.const
+   * val iter = 2.const
+   * x.unbiased(bias, iter).eval should be(Tensor.scalar(2))
+   * }}}
+   *
+   * @param out tensor
+   * @param bias biased component
+   * @param iter computation iteration
+   * @return unbiased value
+   */
+  def unbiased[A: TensorType: Numeric](out: F[A], bias: F[A], iter: F[Int]): F[A]
```

Comment:
That expression is really fixing the fact that the algorithm is `0` biased at first steps and slowly accelerates. But the function itself has nothing to do with `unbiasing`. I would suggest renaming that to something like `boost` which increases the value when `iter` is small and when iter is reaching infinity that function works like identity.